src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -115,19 +115,27 @@ public function hash($path)
      * @param  string  $path
      * @param  string  $contents
      * @param  bool  $lock
-     * @return int
+     * @return int|bool
```

Comment:
Why does this have to change?

Reply:
From the [php docs](http://php.net/manual/en/function.file-put-contents.php) for file_put_contents:

>This function returns the number of bytes that were written to the file, or FALSE on failure.